*** /home/stephan/freshtomato-mips/release/src/router/miniupnpd/genconfig.sh	2019-11-02 16:38:49.469935988 +0100
--- /home/stephan/Dokumente/freshtomato-mips/genconfig.sh	2019-09-20 20:22:48.529996028 +0200
***************
*** 387,395 ****
  		if grep uuid_create /usr/include/uuid.h > /dev/null 2>&1 ; then
  			echo "#define BSD_UUID" >> ${CONFIGFILE}
  		fi
! 		if grep uuid_generate /usr/include/uuid/uuid.h > /dev/null 2>&1 ; then
! 			echo "#define LIB_UUID" >> ${CONFIGFILE}
! 		fi
  		;;
  esac
  
--- 387,395 ----
  		if grep uuid_create /usr/include/uuid.h > /dev/null 2>&1 ; then
  			echo "#define BSD_UUID" >> ${CONFIGFILE}
  		fi
! #		if grep uuid_generate /usr/include/uuid/uuid.h > /dev/null 2>&1 ; then
! #			echo "#define LIB_UUID" >> ${CONFIGFILE}
! #		fi
  		;;
  esac
  
